src/main/java/com/synopsys/integration/detect/lifecycle/run/operation/OperationRunner.java

```diff
@@ -953,15 +956,35 @@ public Optional<File> findRapidScanConfig() throws OperationException {
                 .findRapidScanConfig(directoryManager.getSourceDirectory())
         );
     }
-    
+
+    private int countBdioEntryFiles() throws IntegrationException {
+        File bdioFile = this.fileFinder.findFile(directoryManager.getBdioOutputDirectory(), "*.bdio");
+        Bdio2ContentExtractor bdio2Extractor = new Bdio2ContentExtractor();
+        return bdio2Extractor.extractContent(bdioFile).size() - 1; // excludes bdio header file
+    }
+
+    public int calculateMaxWaitInSeconds(int bdioEntriesCount) {
+        // Max polling interval time will be the (N+1)th Fibonacci number in seconds, where N is the number of BDIO chunks (entry files)
+        int maxWaitInSeconds = 1;
+        int[] customFibonacciSequence = {0, 1, 1, 2, 3, 5, 8, 13, 21, 34, 55};
+
+        if (bdioEntriesCount > customFibonacciSequence.length - 1) {
```

Comment:
After above changes on class field and minWaitInSeconds, this method can be removed. The replacement logic is in watForBomCompletion.